### PR TITLE
Add prettier & lint for Typescript files

### PR DESCRIPTION
**Pull request by thgreasi**
Also removes prettier-eslint-cli (since it was giving errors #259) in favor of the latest prettier & tslint (obviously np to revert that part if we don't like it).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,12 +48,14 @@
...
-    "prettier-eslint-cli": "~4.6.0",
+    "prettier": "^1.9.2",
     "react": "^15.6.2",
     "react-dom": "^15.6.0",
     "rimraf": "^2.6.2",
     "ts-loader": "^3.0.2",
-    "ts-node": "^3.3.0"
+    "ts-node": "^3.3.0",
+    "tslint": "^5.8.0",
+    "typescript": "^2.6.2"
```
Comment (thgreasi) [inline package.json]: So, all this time the TS part of the build was using the local machine's typescript installation!

File: src/components/Alert.tsx [hunk 2]
```diff
@@ -23,11 +24,11 @@ const getTitle = (props: AlertProps) => {
...
-	return props.danger ? 'Darn!' :
-		props.warning ? 'Warning!' :
-		props.success ? 'Great!' :
-		props.info ? 'Hey!' :
-		'';
+	return props.danger
+		? 'Darn!'
+		: props.warning
+			? 'Warning!'
+			: props.success ? 'Great!' : props.info ? 'Hey!' : '';
```
Comment (thgreasi) [inline src/components/Alert.tsx]: I hate how prettier formats multi-ternary expressions.
They have an issue to track this... but still no progress https://github.com/prettier/prettier/issues/737

File: package.json (hunk 1 above)
Comment (LucianBuzzo) [reply package.json]: Doesn't look like it, though that is probably entirely by accident!
```
$ resin-components/ (255-dropdown-color) ll node_modules | grep typescript
drwxr-xr-x   15 lucianbuzzo  staff   510B 20 Dec 17:33 typescript/
drwxr-xr-x    8 lucianbuzzo  staff   272B 20 Dec 17:33 typescript-eslint-parser/
```

I did some digging and it looks like `prettier-eslint` installs typescript as a dependency.

Comment (thgreasi) [reply package.json]: Oh, that's cool(er) then :)

**Pull request merged**